bin/main.ml

```diff
@@ -80,83 +25,199 @@ let gen_output ast f = function
       print_endline res
   | _ -> ()
 
-let process_pragmas (pragmas : Syntax.pragmas) : unit =
-  let process_global_pragma (k, v) =
-    match (k, v) with
-    | Syntax.PrintUsage, _ -> usage () |> print_endline
-    | Syntax.ShowPragmas, _ -> Syntax.show_pragmas pragmas |> print_endline
-    | Syntax.NestedProtocols, _ ->
-        if Option.is_some !fsm then
-          Err.uerr (Err.IncompatibleFlag ("fsm", Syntax.show_pragma k))
+let main file enumerate verbose go_path out_dir project fsm gencode_ocaml
```

Comment:
Calling main sounds hellish, can we do a record of options or something with more structure?

Reply:
It doesn't change the fact that we need to unpack all of them.

Reply:
Calling main is hellish because "main "toto" true false "./foo" "toto" true false" is very confusing. The fact that inside it is unpacked it is not terrible. 

However, having said this, main is not a function that is really called in more than one place, and the one place where it's called already has very descriptive variable names, so this is not really a problem. Maybe we can leave it as is.